core/src/main/java/nl/nn/adapterframework/pipes/Text2XmlPipe.java

```diff
@@ -36,57 +36,58 @@
  * 
  * @author J. Dekker
  */
-public class Text2XmlPipe extends FixedForwardPipe {
-	private String xmlTag;
-	private boolean includeXmlDeclaration = true;
-	private boolean splitLines = false;
-	private boolean replaceNonXmlChars = true;
-	private boolean useCdataSection = true;
-	
+public class Text2XmlPipe extends StreamingPipe {
+	private @Getter String xmlTag;
+	private @Getter boolean includeXmlDeclaration = true;
+	private @Getter boolean splitLines = false;
+	private @Getter boolean replaceNonXmlChars = true;
+	private @Getter boolean useCdataSection = true;
 
 	@Override
 	public void configure() throws ConfigurationException {
 		super.configure();
 
 		if (StringUtils.isEmpty(getXmlTag())) {
-			throw new ConfigurationException("You have not defined xmlTag");
+			throw new ConfigurationException("Attribute [xmlTag] must be specified");
 		}
 	}
-	
-	
+
 	@Override
 	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
 		String result;
-		try {
-			if (isSplitLines() && message != null) {
-				Reader reader = message.asReader();
-				if (replaceNonXmlChars) {
-					reader = new EncapsulatingReader(reader, "", "", true);
-				}
-				BufferedReader br = new BufferedReader(reader);
-
-				String l;
-				StringBuffer buffer = new StringBuffer();
-
-				while ((l = br.readLine()) != null) {
-					buffer.append("<line>"+addCdataSection(l)+"</line>");
+		try (MessageOutputStream target = getTargetStream(session)) {
+			try (Writer writer = target.asWriter()) {
```

Comment:
Considering that the pipe produces XML, it would be more appropriate to send the output to `target.asContentHandler()` instead of `target.asWriter()`